Sources/TuistKit/Services/BuildService.swift

```diff
@@ -70,41 +77,87 @@ final class BuildService {
         } else {
             var cleaned: Bool = false
             // Run only buildable entry schemes when specific schemes has not been passed
             let buildableEntrySchemes = buildGraphInspector.buildableEntrySchemes(graphTraverser: graphTraverser)
             try buildableEntrySchemes.forEach {
-                try buildScheme(scheme: $0, graphTraverser: graphTraverser, path: path, clean: !cleaned && clean, configuration: configuration)
+                try buildScheme(
+                    scheme: $0,
+                    graphTraverser: graphTraverser,
+                    workspacePath: workspacePath,
+                    clean: !cleaned && clean,
+                    configuration: configuration
+                )
                 cleaned = true
             }
         }
 
+        if let productsPath = productsPath {
+            try copyBuildProducts(to: productsPath, workspacePath: workspacePath)
+        }
+
         logger.log(level: .notice, "The project built successfully", metadata: .success)
     }
 
     // MARK: - private
 
-    private func buildScheme(scheme: Scheme, graphTraverser: GraphTraversing, path: AbsolutePath, clean: Bool, configuration: String?) throws {
+    private func buildScheme(
+        scheme: Scheme,
+        graphTraverser: GraphTraversing,
+        workspacePath: AbsolutePath,
+        clean: Bool,
+        configuration: String?
+    ) throws {
         logger.log(level: .notice, "Building scheme \(scheme.name)", metadata: .section)
         guard let (project, target) = buildGraphInspector.buildableTarget(scheme: scheme, graphTraverser: graphTraverser) else {
             throw BuildServiceError.schemeWithoutBuildableTargets(scheme: scheme.name)
         }
-        let workspacePath = try buildGraphInspector.workspacePath(directory: path)!
+
         let buildArguments = buildGraphInspector.buildArguments(project: project, target: target, configuration: configuration, skipSigning: false)
-        _ = try xcodebuildController.build(
-            .workspace(workspacePath),
-            scheme: scheme.name,
-            clean: clean,
-            arguments: buildArguments
-        )
-        .printFormattedOutput()
-        .toBlocking()
-        .last()
+
+        _ = try xcodeBuildController
+            .build(
+                .workspace(workspacePath),
+                scheme: scheme.name,
+                clean: clean,
+                arguments: buildArguments
+            )
+            .printFormattedOutput()
+            .toBlocking()
+            .last()
+    }
+
+    private func copyBuildProducts(
+        to outputPath: AbsolutePath,
+        workspacePath: AbsolutePath
+    ) throws {
+        let buildOutputDirectory = try xcodeBuildLocator.locateProductsDirectory(for: workspacePath)
+
+        guard FileHandler.shared.exists(buildOutputDirectory) else {
+            logger.error("No build products found at \(buildOutputDirectory.pathString), skipping copy products")
```

Comment:
I don't think it's completely right to log an error but not throw an error - I think we can throw an error here instead

Reply:
Agree! Throwing a `.bug` here.